pkg/network/event_common.go

```diff
@@ -421,3 +452,77 @@ func generateConnectionKey(c ConnectionStats, buf []byte, useNAT bool) []byte {
 	n += raddr.WriteTo(buf[n:]) // 4 or 16 bytes
 	return buf[:n]
 }
+
+// Sub returns s-other
+func (s StatCounters) Sub(other StatCounters) (sc StatCounters, underflow bool) {
+	if s.RecvBytes < other.RecvBytes ||
+		s.RecvPackets < other.RecvPackets ||
+		(s.Retransmits < other.Retransmits && s.Retransmits > 0) ||
+		s.SentBytes < other.SentBytes ||
+		s.SentPackets < other.SentPackets ||
+		(s.TCPClosed < other.TCPClosed && s.TCPClosed > 0) ||
+		(s.TCPEstablished < other.TCPEstablished && s.TCPEstablished > 0) {
+		return sc, true
+	}
+
+	sc = StatCounters{
+		RecvBytes:   s.RecvBytes - other.RecvBytes,
+		RecvPackets: s.RecvPackets - other.RecvPackets,
+		SentBytes:   s.SentBytes - other.SentBytes,
+		SentPackets: s.SentPackets - other.SentPackets,
+	}
+
+	if s.Retransmits > 0 {
+		sc.Retransmits = s.Retransmits - other.Retransmits
+	}
+	if s.TCPEstablished > 0 {
+		sc.TCPEstablished = s.TCPEstablished - other.TCPEstablished
+	}
+	if s.TCPClosed > 0 {
+		sc.TCPClosed = s.TCPClosed - other.TCPClosed
+	}
```

Comment:
do we need these conditionals? I thought the early statement was ensuring that the subtraction will never result in a negative number

Reply:
If `s.TCPClosed == 0` above but `s.TCPClosed < other.TCPClosed`, we would still end up down here, so have to protect against that case.